docs/source/recipes/how-to-use-color-picker-widget.md

```diff
@@ -0,0 +1,139 @@
+---
+myst:
+  html_meta:
+    "description": "How to use the color picker widget in blocks settings and forms"
+    "property=og:description": "How to use the color picker widget in blocks settings and forms"
+    "property=og:title": "How to use the color picker widget"
+    "keywords": "Volto, Plone, frontend, React, blocks, forms, widget, color, picker"
+---
+
+# How to use the color picker widget
+
+Volto comes with a color picker widget that can be used in any Volto form.
+The most common use case is use this value along with the blocks StyleWrapper as show in the example. See {ref}`block-style-wrapper-label` for more background about how the blocks StyleWrapper works.
+Combined with the blocks StyleWrapper you can have a powerful, yet simple way to manage color properties in your blocks.
+It can be used in your custom block settings schema or enhance an existing block as follows:
+
+```js hl_lines="13-16, 31-42"
+import { addStyling } from '@plone/volto/helpers/Extensions/withBlockSchemaEnhancer';
+import { defineMessages } from 'react-intl';
+import config from '@plone/volto/registry';
+
+const messages = defineMessages({
+  backgroundColor: {
+    id: 'Background color',
+    defaultMessage: 'Background color',
+  },
+});
+
+export const defaultStylingSchema = ({ schema, formData, intl }) => {
+  const BG_COLORS = [
+    { name: 'transparent', label: 'Transparent' },
+    { name: 'grey', label: 'Grey' },
+  ];
+
+  // You could allow passing the color definition from the config or from the default
+  // defined above
+  const colors =
+    config.blocks?.blocksConfig?.[formData['@type']]?.colors || BG_COLORS;
+
+  // Same for the default used (or undefined)
+  const defaultBGColor =
+    config.blocks?.blocksConfig?.[formData['@type']]?.defaultBGColor;
+
+  // This adds the StylingWrapper support to your block
+  addStyling({ schema, intl });
+
+  // Then we add the field to the fieldset inside the StylingWrapper `styles` field schema fieldset array
+  schema.properties.styles.schema.fieldsets[0].fields = [
+    ...schema.properties.styles.schema.fieldsets[0].fields,
+    'backGroundColor',
+  ];
+
+  // and finally, we add the field to the StylingWrapper `styles` object field schema properties
+  schema.properties.styles.schema.properties['backGroundColor'] = {
+    widget: 'color_picker',
+    title: intl.formatMessage(messages.backgroundColor),
+    colors,
+    default: defaultBGColor,
+  };
+
+  return schema;
+};
+```
+
+The widget name the color picker widget uses is `color_picker`.
+
+## Color definitions
+
+The `colors` property of the widget controls which colors are available to choose in the widget.
+This is the signature of the object:
+
+```ts
+type Color =
+  | {
+      label: string;
+      style: Record<`--${string}`, string> & { name: string };
+    }
+  | {
+      name: string;
+      label: string;
+    };
+
+const colors: Color[] = [
+  { style: { name: 'red', '--background-color': 'red' }, label: 'red' },
```

Comment:
Why is `name` included as part of `style` when it's not a CSS property?

Reply:
I know, I'm not 100% sure too... We need a string value still, for backwards compatibility since we need it here and there in the rendered HTML in the widget. It also serves a double purpose, since it also turns into an injected class name, in case you need it too. The code to comply with BW compat was also simpler, so I left it there. We could lift it to the main object, although and if you still need the class name for some reason, then manually add it.

Reply:
Oh, it's a class name? That would be more obvious if it were called `className`. It still feels a bit weird to me to have it inside `style`, but if it's needed for backwards compatibility...

Reply:
not a class name, but as it's not a custom CSS property (it does not start with `--`), it comes for free... because of the stylewrapper (which takes all what's inside and does its thing). It's more a convention... I'd be ok and try to move it to top. I can work on this these days.

Reply:
@davisagli I've moved the `name` key to the top, it makes more sense as you said, and the benefits are not that much in the other form. Please, take a look!

Reply:
@giuliaghisini I've modified a bit the definitions as per David's request, please take another quick look!